molecule/mysql/prepare.yml

```diff
@@ -0,0 +1,121 @@
+---
+- name: Prepare
+  hosts: all
+
+  vars:
+    inventory_dir: "{{ lookup('env', 'MOLECULE_SCENARIO_DIRECTORY') }}/../../environments/vm"
+    copy_tls_ca:
+      - src: /root/CA.crt
+        dest: "{{ inventory_dir }}/files/certs/galera/galera_ca.pem"
+      - src: /root/CA_key.pem
+        dest: "{{ inventory_dir }}/files/certs/galera/galera_ca.key"
+
+  pre_tasks:
+    - name: Read vars from secrets file
+      include_vars: "../../environments/vm/secrets/vm.yml"
+    - name: Read vars from group_vars file
+      include_vars: "../../environments/vm/group_vars/vm.yml"
+
+    - name: "Install package(s)"
+      yum:
+        name:
+          - crontabs
+          - python2-pip
+        state: present
+
+    - name: Install python crypto module
+      pip:
+        name: "cryptography"
+        state: "latest"
+      run_once: true
+
+    - name: create CA key
+      openssl_privatekey:
+        path: /root/CA_key.pem
+      register: ca_key
+      run_once: true
+
+    - name: create the CA CSR
+      openssl_csr:
+        path: /root/CA.csr
+        privatekey_path: "{{ ca_key.filename }}"
+        common_name: "my-ca"
+      register: ca_csr
+      run_once: true
+
+    - name: sign the CA CSR
+      openssl_certificate:
+        path: /root/CA.crt
+        csr_path: "{{ ca_csr.filename }}"
+        privatekey_path: "{{ ca_key.filename }}"
+        provider: selfsigned
+      register: ca_crt
+      run_once: true
+
+    - name: "Fetch CA crt"
+      fetch:
+        src: "{{ item.src }}"
+        dest: "{{ item.dest }}"
+        flat: true
+      with_items: "{{ copy_tls_ca }}"
+      run_once: true
+
+    - name: "Copy CA crt file"
+      copy:
+        src: "{{ item.dest }}"
+        dest: "{{ item.src }}"
+      with_items: "{{ copy_tls_ca }}"
+
+    - name: create host CSR signing key
+      openssl_privatekey:
+        path: /root/example_com_host_key.pem
+      register: example_com_key
+
+    - name: create the CSR for the LDAP server
```

Comment:
LDAP should be Galera?

Reply:
Ah yes, was a copy/paste. Will update it with next commit. 👍